src/decorators/types.js

```diff
@@ -1,9 +1,15 @@
 // @flow
 
+export type Cache = {
+  fnName: string,
+  keySets: Array<Array<string>>,
+  memoized: Function,
+};
+
 export type Decorator<TType> = (
-  target: TType,
-  name: $Keys<TType>,
   descriptor: Descriptor,
```

Comment:
You might need to change `export type Decorator<TType: Object>` but I hope you don't need this